internals/search.py

```diff
@@ -459,7 +459,7 @@ def process_query(
   result_id_set.intersection_update(permission_ids)
   logging.info('got %r result IDs with permissions', len(result_id_set))
 
-  result_id_list = list(result_id_set)
+  result_id_list = feature_helpers.filter_confidential_ids(list(result_id_set))
```

Comment:
This line also looks like it would be too slow.  Rather than post-filtering, the permission check should be part of `permission_terms` query that runs in parallel with the user-entered query terms.  There is already logic there for excluding unlisted feature entries.  However, there is a TODO comment for adding back personalized results (the unlisted items that the current user is actually allowed to see).  Would it be OK to also leave that aspect as a TODO for confidential results at first?